Added support for different parity settings and YTL meter

arjanmels opened the pull request:
- Added support for different parity settings for different meters
- Added support for using modbus function code 3 in addition to 4
- Added support for YTL DTS353F-2 kWh meter
- Enabled MCLR pin (to allow resetting via ICD connector)
- Fixed superfluous ;
- Cleaned automatically generated files from repo
- Added phase currents to display
- Error detection for all meters
- Display Connected for STATE_B

arjanmels committed:
Removed erroneous ;

arjanmels committed:
Enabled reset via MCLR pin

arjanmels committed:
Clean up

arjanmels committed:
Add compile time option for modbus parity bits

arjanmels committed:
Modbus function 4=>3

arjanmels committed:
Added support for multiple parity settings and YTL meter

mstegen reviewed (commented):
Hi Arjan, 
I don't think this will work, as the config bytes are pre-set when flashing the controller for the first time, and should not be modified with the bootloader.
Also, as there is no external pull up on the MCLR line, setting it to external while noting is connected to the pin, can cause resets. (even with the internal weak pull up)

mstegen reviewed (approved):
Good catch, thanks.

arjanmels commented:
> I don't think this will work, as the config bytes are pre-set when flashing the controller for the first time, and should not be modified with the bootloader.

You're right, I did a flash with a pickit to set this. It is not a must have as you can enter and exit bootloader via protocol. I added it because I was stuck once or twice in the bootloader during debugging (by faulty baudrate detection I guess.).

> Also, as there is no external pull up on the MCLR line, setting it to external while noting is connected to the pin, can cause resets. (even with the internal weak pull up)

Ok, I had expected the internal pull-up to be sufficient.

I'll remove it in my next push. BTW still running some tests, hence the draft status.

arjanmels committed:
Improvements for EV Meter
- Fixed YTL byte order
- Added phase currents to display
- Error detection for all meters
- Display Connected for STATE_B

arjanmels committed:
Reverted to INTMCLR

arjanmels commented:
Added further improvements and fixes and tested properly.

arjanmels committed:
Corrected EnergyCharged display
